API/ev3_timer.h

```diff
@@ -172,6 +178,60 @@ void SetTimerUS(uint8_t Timer, unsigned long Value);
  */
 bool SetTimerCallback(TimerInterval interval, TimerCallback callback);
 
+/*!
+ * \brief Create a new timer mutex.
+ *
+ * This mutex is ONLY usable for synchronization between
+ * a single thread and a peridic callback handler.
+ *
+ * \param ppMutex On success, this pointer will be filled with a pointer to the new mutex.
+ * \return True if the operation succeeded, false otherwise.
+ */
+bool TimerCreateMutex(TimerMutex **ppMutex);
+
+/*!
+ * \brief Destroy an existing timer mutex.
+ * \param pMutex Mutex to destroy.
+ * \return True if the operation succeeded, false otherwise.
+ */
+bool TimerDestroyMutex(TimerMutex *pMutex);
+
+/*!
+ * \brief Try to lock a given timer mutex.
+ *
+ * This mutex is ONLY usable for synchronization between
+ * a single thread and a peridic callback handler.
+ *
+ * \param pMutex Mutex to try to lock.
+ * \return True if the caller now holds the lock, false otherwise.
+ */
+bool TimerTryLockMutex(TimerMutex *pMutex);
+
+/*!
+ * \brief Lock a given timer mutex.
+ *
+ * This mutex is ONLY usable for synchronization between
+ * a single thread and a peridic callback handler.
+ *
+ * After this function exits, the caller will be holding this lock.
+ *
+ * Beware! If you are not careful, this could result in a deadlock.

```

Comment:
If you mask the signals while the mutex is locked, you will rule out the possibility of deadlock.

Reply:
This got resolved by abandoning signals :D